pkg/operator/workloads/workflow.go

```diff
@@ -280,12 +281,13 @@ func IsWorkloadEnded(appName string, workloadID string) (bool, error) {
 	return false, errors.New("workload not found in the current context")
 }
 
-func IsDeploymentUpdating(appName string) (bool, error) {
+func GetDeploymentStatus(appName string) (resource.DeploymentStatus, error) {
 	ctx := CurrentContext(appName)
 	if ctx == nil {
-		return false, nil
+		return resource.UnknownDeploymentStatus, nil
 	}
 
+	updatingCount := 0
 	for _, workload := range extractWorkloads(ctx) {
 
 		// Pending HPA workloads shouldn't block new deployments
```

Comment:
nit: update this comment, maybe "HPA workloads don't really count" or something similar...